### PR TITLE
feat: Implement public API that will open SpaceBindings to end users through proxy part 2

### PR DESCRIPTION
**Pull request by mfrancisc**
Update `TestSpaceLister` so that it verifies if the `workspace.Status.Bindings` field is populated correctly.

Jira: https://issues.redhat.com/browse/ASC-410

Paired with:
- registration-service: https://github.com/codeready-toolchain/registration-service/pull/350

Related PRs:
- toolchain-common: https://github.com/codeready-toolchain/toolchain-common/pull/329

### CODE REVIEW

File: test/e2e/proxy_test.go [hunk 1]
```diff
@@ -693,7 +695,43 @@ func TestSpaceLister(t *testing.T) {
 
 			// then
 			require.NoError(t, err)
-			verifyHasExpectedWorkspace(t, expectedWorkspaceFor(t, awaitilities.Host(), users["car"], commonproxy.WithType("home"), appStudioTierRolesWSOption), *workspace)
+			verifyHasExpectedWorkspace(t, expectedWorkspaceFor(t, awaitilities.Host(), users["car"], commonproxy.WithType("home"), appStudioTierRolesWSOption,
+				commonproxy.WithBindings([]toolchainv1alpha1.Binding{
+					{MasterUserRecord: "bus", Role: "admin", AvailableActions: []string{"update", "delete"}},
+					{MasterUserRecord: "car", Role: "admin", AvailableActions: []string{"update", "delete"}},
+					{MasterUserRecord: "road-bicycle", Role: "admin", AvailableActions: []string{"update", "delete"}}},
```
[the remark this answers is not in the document]
Comment (mfrancisc) [reply test/e2e/proxy_test.go:702]: yup, let me update the tests here! thanks!

[the remark this answers is not in the document]
Comment (mfrancisc) [reply test/e2e/proxy_test.go]: Done in https://github.com/codeready-toolchain/toolchain-e2e/pull/805/commits/500d8ca0098ff0eeb0b4f8dd756212dba8c7fd08

The `shareSpaceWith` function now uses SBR to share workspaces.

File: test/e2e/proxy_test.go [hunk 2]
```diff
@@ -693,7 +695,43 @@ func TestSpaceLister(t *testing.T) {
...
+				testspace.WithSpecTargetCluster(memberAwait.ClusterName),
+				testspace.WithLabel(toolchainv1alpha1.SpaceCreatorLabelKey, "car"), // atm this label is used by the spacelister to set the owner of the workspace.
+			)
+			subSpace, _ = testsupportspace.VerifyResourcesProvisionedForSpace(t, awaitilities, subSpace.Name)
+			// override username so that it uses the name of the sub-space when creating the expected workspace.
+			carUser := *users["car"]
+			carUser.compliantUsername = subSpace.GetName()
+
+			// when
+			workspace, err := users["car"].getWorkspace(t, hostAwait, subSpace.GetName())
```
[the remark this answers is not in the document]
Comment (mfrancisc) [reply test/e2e/proxy_test.go]: Good point! I'm now getting the subSpace also using the other users ("bicycle" and "bus").

I've also fixed the expected Bindings list as it was incorrect in some assertions, and added the binding request field.

Please see: https://github.com/codeready-toolchain/toolchain-e2e/pull/805/commits/500d8ca0098ff0eeb0b4f8dd756212dba8c7fd08

File: test/e2e/proxy_test.go [hunk 3]
```diff
@@ -55,13 +57,18 @@ type patchStringValue struct {
...
-	primaryUserSpace, err := hostAwait.WaitForSpace(t, u.compliantUsername, wait.UntilSpaceHasAnyTargetClusterSet(), wait.UntilSpaceHasAnyTierNameSet())
+	primaryUserSpace, err := awaitilities.Host().WaitForSpace(t, u.compliantUsername, wait.UntilSpaceHasAnyTargetClusterSet(), wait.UntilSpaceHasAnyTierNameSet(), wait.UntilSpaceHasAnyProvisionedNamespaces())
 	require.NoError(t, err)
-	CreateSpaceBinding(t, hostAwait, guestUserMur, primaryUserSpace, "admin") // creating a spacebinding gives guestUser access to primaryUser's space
+	spaceBindingRequest := CreateSpaceBindingRequest(t, awaitilities, primaryUserSpace.Spec.TargetCluster,
+		WithSpecSpaceRole("admin"),
+		WithSpecMasterUserRecord(guestUserMur.GetName()),
+		WithNamespace(testsupportspace.GetDefaultNamespace(primaryUserSpace.Status.ProvisionedNamespaces)),
+	)
+	return spaceBindingRequest
```
Comment (MatousJobanek) [inline test/e2e/proxy_test.go:71]: let's wait for the SpaceBinding to be created
```suggestion
	_, err = awaitilities.Host().WaitForSpaceBinding(t, guestUserMur.GetName(), primaryUserSpace.GetName())
	require.NoError(t, err)
	return spaceBindingRequest
```

**Pull request merged**